### ISSUE TITLE
Question about the precision of checkpoint

### ISSUE THREAD

**Issue by sh0416** [facebookresearch/llama#216]
Hello.

I am wondering what kind of precision strategy is applied during the pretraining.

I couldn't find out the precision, e.g., fp16, bf16, or full fp32, in the paper.

The only clue is the dtype of state dict inside the checkpoint, which is fp16. 

To the best of our knowledge, even though we use the mixed precision training, using full precision checkpoint is the best practice.

I think it would be mixed precision with fp16 or the whole checkpoint would be managed by fp16.

Which one is correct? or is there any other strategy for me to check?

Also, it is really interesting for me that the loss curve of LLaMA is really stable, which is not found in OPT case.

FP16 could be the one factor to cause the unstability, so could you explain what happened after OPT..?

Thank you in advance

**Comment by richarddwang** [comment]
Hi, @sh0416 
Did you figure out which precision should be used to load the checkpoints?

**Comment by nenkoru** [comment]
+1 for a question

**Comment by imoneoi** [comment]
@sh0416 Any updates? Did you figure out the precision used in pretraining (FP16, bfloat16, TF32 or FP32)?

**Comment by tongyx361** [comment]
> Use FP16

May I ask where you found this answer? Could you be so kind to provide some corresponding references?

**Comment by sh0416** [comment]
In my experience, most checkpoints works well with FP16 weights.
Sometimes, the loss from precision could occurs, but most of them come from the normalization weight.
So, my conclusion is use fp16 weights and something feels weird, then use fp32 normalization weight or full fp32 normalization weight and comparing the result with fp32 version.
My conclusion is clearly my subjective opinion and there is no guarantee about taking this precision strategy and I will not take any credit for the risk by taking my decision. However, give it a try.
I am pretty certain that your experience will give you some confidence about using this strategy and the overall process of LLM.

**Comment by sh0416** [comment]
FYI, all comparisons between fp32 and fp16 due to my weird feeling ends up to the same result except the normalization weight. Also, consider that modern accelerator process fp16 values at most 3 times faster than processing fp32 values.

**Comment by tongyx361** [comment]
Thank you a lot over for such a detailed and patient answer!
